File: 0617/SeoyeonHong/개인정보 수집 유효기간.py
# https://school.programmers.co.kr/learn/courses/30/lessons/150370?language=python3
# 파기해야 할 개인정보 번호(오름차순)
def date_num(date):
    year, month, day = list(map(int, date.split('.')))
    num = year * 12 * 28 + month * 28 + day
    return num
    
def solution(today, terms, privacies):
    answer = []
    today_num = date_num(today)
    type_num = {}
    for term in terms:
        term_type, month = term.split()
        type_num[term_type] = int(month) * 28
    
    
    for i in range(len(privacies)):
        privacy = privacies[i]
        date, term_type = privacy.split()
        term_num = date_num(date)
        if today_num - term_num >= type_num[term_type]:
            answer.append(i+1)
            
    return answer

File: 0617/SeoyeonHong/개인정보 수집 유효기간.java
import java.util.ArrayList;
import java.util.HashMap;

class Solution {
    public static int days_of_date(String date){
        String[] str = date.split("\\.");
        int number = Integer.parseInt(str[0]) * 12 * 28 + Integer.parseInt(str[1]) * 28 + Integer.parseInt(str[2]);
        return number;
    }

    public int[] solution(String today, String[] terms, String[] privacies) {
        int[] answer = {};
        int days_of_today = days_of_date(today);
        HashMap<String, Integer> days_of_type = new HashMap<>();
        for(int i = 0; i < terms.length; i++) {
            String[] str = terms[i].split(" ");
            days_of_type.put(str[0], Integer.parseInt(str[1]) * 28);
        }
        ArrayList<Integer> list = new ArrayList<>();
        for(int i = 0; i < privacies.length; i++){
            String[] str = privacies[i].split(" ");
            int days_of_term = days_of_date(str[0]);
            
            if(days_of_today - days_of_term >= days_of_type.get(str[1])){
                list.add(i+1);
            }
        }

        answer = new int[list.size()];
        for(int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}